Convert to `rattler-build`

jaimergp opened the pull request.

jaimergp commented:
This is ready but I need consensus on the dependency split inconsistencies mentioned in https://github.com/napari/packaging/issues/261

jaimergp commented on conda-recipe/recipe.yaml:
```diff
@@ -0,0 +1,211 @@
+        - pydantic >=2.2.0
+        - pygments >=2.6.0
+        - pyopengl >=3.1.5
+        - python-dotenv >=0.21.0
```
```suggestion
```

willingc commented on .github/workflows/make_bundle_conda.yml:
```diff
@@ -43,6 +43,15 @@ on:
+concurrency:
+  # Concurrency group that uses the workflow name and PR number if available
+  # or commit SHA as a fallback. If a new build is triggered under that
+  # concurrency group while a previous build is running it will be canceled.
```
```suggestion
  # concurrency group, while a previous build is running, the previous build will be canceled.
```

willingc commented on .github/workflows/make_bundle_conda.yml:
```diff
@@ -79,7 +88,8 @@ jobs:
         with:
           fetch-depth: 0
-          repository: conda-forge/napari-feedstock
+          repository: jaimergp/napari-feedstock  # REVERT to conda-forge
```
Is this a TODO?

Czaki replied:
I think that it is waiting on https://github.com/conda-forge/napari-feedstock/pull/87

willingc commented on conda-recipe/recipe.yaml:
```diff
@@ -0,0 +1,209 @@
+              - 'QT_DEBUG_PLUGINS=1 LIBGL_DEBUG=verbose xvfb-run --auto-servernum napari --info | grep -C99 "napari: ${PKG_VERSION}"'
+            else:
+              - napari --info
+          # - pytest --pyargs napari
```
Perhaps add a comment above this line: If the following line is uncommented, also uncomment requirements.

willingc reviewed (commented):
Thanks @jaimergp. Love the conversion.

jaimergp commented:
The CI that builds the installers was failing because we were building conda packages with `v0.6.1.dev140` but installers with `v0.6.1dev140` (notice the dot before `dev`). This created flaky solver issues that sometimes would pass, sometimes wouldn't. I fixed this inconsistency now so we always use dotless versions (due to https://github.com/conda/conda/issues/12568).

Then I ran into `xvfb-run` segfaults, most likely caused by the Docker setup used to build the conda package (which I might remove in the future because we don't need that kind of isolation, but I still need to think about it). I replaced it with a simple `napari --version` there, we are running `napari --info` later in the installer tests anyway.
